### PR TITLE
Starlark Filter support

### PR DESCRIPTION
**Pull request by pwittrock**
Wrapper to support implementing kio.Filters in Starlark

### CODE REVIEW

**Comment by pwittrock** [comment]
@mengqiy

File: kyaml/starlark/doc.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package starlark contains a kio.Filter which can be applied to resources to transform
```
Comment (mengqiy) [inline kyaml/starlark/doc.go]: Probably it also worth to talk about why starlark is picked as a native supported language.

File: kyaml/starlark/starlark.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+)
+
+// Filter transforms a set of resources through the provided program
+type Filter struct {
+	Name string
+
+	// Program is a starlark script which will be run against the resources
+	Program string
+
+	// URL is the url of a starlark program to fetch and run
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply kyaml/starlark/starlark.go]: Good question -- I think as long as folks are aware of that they are using a URL and can make the decision no to, then this is ok.

The [spec](https://github.com/google/starlark-go/blob/master/doc/spec.md) doesn't seem to expose things like writing to the file system, or making sys calls.  The biggest issue would be making modifications to the resources, which could be reviewed is using a gitops approach.

File: kyaml/starlark/doc.go (hunk 1 above)
Comment (pwittrock) [reply kyaml/starlark/doc.go]: Done

**Comment by mengqiy** [comment]
/lgtm
/hold
Hold it for a while in case @monopole want to take a look.
@pwittrock Feel free to remove the hold label if you have sync'd with @monopole

**Comment by mengqiy** [comment]
Chatted with @pwittrock, `kyaml` directory is a package that is independent of kustomize, we can just this PR in.
/hold cancel

**Comment by errordeveloper** [comment]
It was interesting to discover this, I recall posting #438 a while back!

**Pull request merged**